blocks/api/registration.js

```diff
@@ -113,10 +113,21 @@ export function registerBlockType( name, settings ) {
 	if ( ! settings.icon ) {
 		settings.icon = 'star-filled';
 	}
+
+	const attributes = settings.attributes
+		? settings.attributes
+		: get( window._wpBlocksAttributes, name );
+
 	const block = blocks[ name ] = {
-		name,
-		attributes: get( window._wpBlocksAttributes, name ),
 		...settings,
+		name,
+		attributes: keys( attributes ).reduce( ( memo, attributeKey ) => {
```

Comment:
From your previous comment:

> I see value in leaving the matchers as the non-explicit ones to avoid listing all the matchers.

Is the whole existence of this `reduce` here to support being able to specify the `comment` case?

Reply:
yes?